crates/core/component/dex/src/state_key.rs

```diff
@@ -27,6 +27,39 @@ pub fn all_positions() -> &'static str {
     "dex/position/"
 }
 
+pub mod candlesticks {
+    use penumbra_proto::DomainType;
+
+    use crate::DirectedTradingPair;
+
+    pub fn block_executions() -> &'static str {
+        "dex/block_executions"
+    }
+
+    pub fn block_position_executions() -> &'static str {
+        "dex/block_position_executions"
+    }
+
+    pub fn block_swap_executions() -> &'static str {
+        "dex/block_swap_executions"
+    }
+
+    pub fn in_progress(pair: &DirectedTradingPair) -> String {
+        let pair_bytes = hex::encode_upper(pair.encode_to_vec());
+        format!("dex/in_progress_candlestick/{pair_bytes}")
+    }
+
+    pub fn by_pair_and_height(pair: &DirectedTradingPair, height: u64) -> String {
+        let pair_bytes = hex::encode_upper(pair.encode_to_vec());
```

Comment:
would prefer if we encoded directed trading pairs consistently across the dex component state keys, `start` + `/` + `end` unless there's a specific reason to do it specially here